### PR TITLE
[AKS] Update the aks install/remove connector to use virtual-kubelet

### PR DESCRIPTION
**Pull request by robbiezhang**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 1]
```diff
@@ -316,7 +316,7 @@ def k8s_install_connector(client, name, resource_group, connector_name,
                           chart_url=None, os_type='Linux'):
     from subprocess import PIPE, Popen
     helm_not_installed = "Helm not detected, please verify if it is installed."
-    image_tag = 'latest'
+    node_prefix = 'virtual-kubelet-' + connector_name.lower()
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: Will this work if connector_name contains spaces or other special chars?
If not, handle this in _params.py where you can add a validator to that param.
Same for uninstall.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 2]
```diff
@@ -343,19 +343,33 @@ def k8s_install_connector(client, name, resource_group, connector_name,
     # Get the TenantID
     profile = Profile()
     _, _, tenant_id = profile.get_login_credentials()
+    # Check if we want the linux connector
+    if os_type.lower() in ['linux', 'both']:
+        _helm_install_aci_connector(url_chart, connector_name, service_principal, client_secret,
+                                    subscription_id, tenant_id, rgkaci.name, location,
+                                    node_prefix + '-linux', "Linux")
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: Typically we prefer single quote strings unless double quote is needed, like if the string actually contains a single quote.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 3]
```diff
@@ -338,68 +338,87 @@ def k8s_install_connector(client, name, resource_group, connector_name,
...
     if os_type.lower() in ['windows', 'both']:
-        # The current connector will deploy two connectors, one for Windows and another one for Linux
-        image_tag = 'canary'
-    logger.warning('Deploying the aci-connector using Helm')
+        _helm_install_aci_connector(url_chart, connector_name, service_principal, client_secret,
+                                    subscription_id, tenant_id, rgkaci.name, location,
+                                    node_prefix + '-win', 'Windows')
+
+def _helm_install_aci_connector(url_chart, connector_name, service_principal, client_secret,
+                                  subscription_id, tenant_id, aciResourceGroup, aciRegion,
```
Comment (mboersma) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: Normally python variables wouldn't use camelCase, but if pylint didn't complain about `aciResourceGroup` then no worries.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_validators.py [hunk 4]
```diff
@@ -90,3 +90,16 @@ def validate_linux_host_name(namespace):
...
                        'letters, numbers, or dashes (-).')
+
+def validate_connector_name(namespace):
+    """Validates a string as a legal connector name.
+
+    This validation will also occur server-side in the kubernetes, but that may take
+    for a while. So it's more user-friendly to validate in the CLI pre-flight.
+    """
+    # https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture/identifiers.md
+    regex = re.compile(r'^[a-z0-9]([a-z0-9\-]*[a-z0-9])?$')  # pylint:disable=line-too-long
```
Comment (mboersma) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_validators.py]: In this project's `.flake8` and `pylintrc` files we set max-line-length to 120. So it seems like this disable comment wouldn't be needed (and some lines below don't need to be wrapped). No need to change anything, I was just curious.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py (hunk 3 above)
Comment (mboersma) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: Actually I think there's an "inconsistent indentation" failure here that one of the linters caught.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_validators.py (hunk 4 above)
Comment (derekbekoe) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_validators.py]: Please remove the `pylint:disable=line-too-long` here.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py (hunk 3 above)
Comment (derekbekoe) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: Please use `aci_resource_group` and `aci_region`.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 5]
```diff
@@ -338,68 +338,87 @@ def k8s_install_connector(client, name, resource_group, connector_name,
...
+                                  nodeName, nodeOsType):
+    image_tag = 'latest'
+    node_taint = 'azure.com/aci'
+    helm_release_name = connector_name.lower() + "-" + nodeOsType.lower()
+    logger.warning("Deploying the ACI connector for '" + nodeOsType + "' using Helm")
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: Change to `logger.warning("Deploying the ACI connector for '%s' using Helm", nodeOsType)`.
Also, can you make `nodeOsType` underscore cased?

**Comment by robbiezhang** [comment]
thanks for the feedbacks. address them and fix all pylint and flake8 violations

**Pull request merged**